### PR TITLE
libgdata: add gnome-online-accounts to propagatedBuildInputs

### PR DESCRIPTION
**Pull request by lightbulbjim**
###### Motivation for this change

`libgdata`'s new Meson config lists `gnome-online-accounts` as a dependency. Downstream apps which use Meson (like `shotwell`) refuse to build without `gnome-online-accounts` in the build env.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
I'm suspect that the error
```
meson.build:117:0: ERROR: Could not generate cargs for libgdata:
```

that line being
```meson
gdata = dependency('libgdata')
```

means that I should probably fix something in `libgdata`.

---

It seems the changes in the `.pc` files are pertinent

Here's a diff of the two files
```diff
--- libgdata-old.pc	2019-07-26 20:48:19.032408179 -0400
+++ libgdata-new.pc	2019-07-26 20:48:44.987912414 -0400
@@ -1,13 +1,14 @@
-prefix=/nix/store/hyakiglm5s4xw24jj001yq51pxglbb3q-libgdata-0.17.9
+prefix=/nix/store/mg16sjmcq5k2nbq63ni9q118a7x4hd5v-libgdata-0.17.10
+libdir=${prefix}/lib
+includedir=/nix/store/icz2m06whc3lwlx1pwgbqpqx9p6irky9-libgdata-0.17.10-dev/include
+
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
-includedir=${prefix}/include
 goa_enabled=true
 
 Name: libgdata
 Description: GData client library
-Version: 0.17.9
-Requires: gobject-2.0 glib-2.0 >= 2.44.0 gio-2.0 >= 2.44.0 libxml-2.0 libsoup-2.4 >= 2.42.0 json-glib-1.0 >= 0.15 
-Requires.private: gthread-2.0 oauth >= 0.9.4 gcr-base-3 libxml-2.0 goa-1.0 >= 3.8
+Version: 0.17.10
+Requires: gobject-2.0, glib-2.0 >=  2.44.0, gio-2.0 >=  2.44.0, gio-unix-2.0, json-glib-1.0 >=  0.15, libxml-2.0, libsoup-2.4 >=  2.42.0, goa-1.0 >=  3.8
+Requires.private: gthread-2.0, oauth >=  0.9.4, gcr-base-3
 Libs: -L${libdir} -lgdata
 Cflags: -I${includedir}/libgdata
```

It seems at some point goa was intentionally private
* https://gitlab.gnome.org/GNOME/libgdata/commit/b4d828e002adf21a5ef671ec1b62bed2f48382af

I guess this should be done again upstream for the generated `.pc` file with meson.

**Comment by worldofpeace** [comment]
Ahh that was an intentional change 
* https://gitlab.gnome.org/GNOME/libgdata/merge_requests/6/diffs?commit_id=0d463dcda944dedce3050f4d541fbfeccba9b569

We should add `gnome-online-accounts` to `propagatedBuildInputs` in `libgdata`.

**Comment by lightbulbjim** [comment]
> We should add `gnome-online-accounts` to `propagatedBuildInputs` in `libgdata`.

Oh yeah, that's a better approach. I'll start testing that; note that it might take me a while to complete test builds of all the reverse dependencies.

**Comment by lightbulbjim** [comment]
Updated to export `gnome-online-accounts` from `libgdata` via `propagatedBuildInputs`. Should cover all reverse dependencies.

**Comment by worldofpeace** [comment]
LGTM, could you fetchpatch the commits off master from these PR's ?
* https://gitlab.gnome.org/GNOME/libgdata/merge_requests/9
* https://gitlab.gnome.org/GNOME/libgdata/merge_requests/10
* https://gitlab.gnome.org/GNOME/libgdata/merge_requests/11

This is something I forgot to do initially.

File: pkgs/development/libraries/libgdata/default.nix [hunk 1]
```diff
@@ -55,6 +55,7 @@ stdenv.mkDerivation rec {
   ];
 
   propagatedBuildInputs = [
+    gnome3.gnome-online-accounts
```
Comment (worldofpeace) [inline pkgs/development/libraries/libgdata/default.nix]: Probably should remove this from `buildInputs` then.

Comment (lightbulbjim) [reply pkgs/development/libraries/libgdata/default.nix]: Good point, updated.

**Comment by lightbulbjim** [comment]
> LGTM, could you fetchpatch the commits off master from these PR's ?
> 
>     * https://gitlab.gnome.org/GNOME/libgdata/merge_requests/9
> 
>     * https://gitlab.gnome.org/GNOME/libgdata/merge_requests/10
> 
>     * https://gitlab.gnome.org/GNOME/libgdata/merge_requests/11
> 
> 
> This is something I forgot to do initially.

I'll take a look.

**Comment by lightbulbjim** [comment]
Added the Meson patches. They seem to work ok and all reverse-dependencies build.

**Comment by lightbulbjim** [comment]
> Added the Meson patches. They seem to work ok and all reverse-dependencies build.
> 
> EDIT: Oops, hang on, I used patches from the merge requests. Let me change it to use the master commits in case the MRs disappear.

Fixed, the Meson fixes now use the master commits.

**Review by worldofpeace — APPROVED** [review]
Thank you so much :sparkles: 

Shotwell now builds locally, everything committed has been done correctly.

**Pull request merged**